### PR TITLE
Walker async improvement

### PR DESCRIPTION
**Pull request by gbalduzz**
Depends on #34.
Slight performance improvement of the CT-AUX walker using async memory transfers.

### CODE REVIEW

**Comment by biddisco** [comment]
Rebased onto master to reduce diffs

**Comment by biddisco** [comment]
test this please

File: include/dca/phys/dca_step/cluster_solver/ctaux/ctaux_walker.hpp [hunk 1]
```diff
@@ -441,6 +441,7 @@ void CtauxWalker<device_t, parameters_type, MOMS_type>::initialize() {
 
 template <dca::linalg::DeviceType device_t, class parameters_type, class MOMS_type>
 void CtauxWalker<device_t, parameters_type, MOMS_type>::do_sweep() {
+  profiler_type profiler("do_sweep", "CT-AUX walker", __LINE__, thread_id);
```
Comment (biddisco) [inline include/dca/phys/dca_step/cluster_solver/ctaux/ctaux_walker.hpp]: You added the profiler in do_sweep, but commented it out in many other places. 
Was that intentional, or just a case of forgetting yo remove the commented out ones before committing?

Comment (gbalduzz) [reply include/dca/phys/dca_step/cluster_solver/ctaux/ctaux_walker.hpp]: No, it is intentional. This function is lengthy and I want it profiled.

File: src/math/nfft/dnfft_1d_kernels.cu [hunk 2]
```diff
@@ -97,7 +97,7 @@ void accumulateOnDevice(const double* M, const int ldm, const int sign, ScalarTy
                         const ScalarType* cubic_coeff, const int size, cudaStream_t stream_) {
   const auto& helper = HelperSelector<ScalarType>::value;
   const static int convolution_size = 2 * helper.get_oversampling() + 1;
-  const auto blocks = getBlockSize(size * size * convolution_size, 128);
+  const auto blocks = getBlockSize(size * size * convolution_size, 85);
```
Comment (biddisco) [inline src/math/nfft/dnfft_1d_kernels.cu]: Could you add a comment to say where the magic number of 85 comes from

Comment (gbalduzz) [reply src/math/nfft/dnfft_1d_kernels.cu]: Sorry, I was playing around with different multiples of conviultion_size (17 * 5 = 85), and forgot it there. There was not any measurable benefit.
Reverted it to 128 in 82b6ec5fe537316af981c1df0d884e514a0d68f2

**Comment by uhaehner** [comment]
retest this please

**Pull request merged**